src/Response/TemporaryFileSapiEmitter.hack

```diff
@@ -37,15 +37,18 @@ final class TemporaryFileSapiEmitter implements EmitterInterface {
 		/* HH_FIXME[4053] */
 		$path = $body->getPath();
 
-		$path as Filesystem\Path;
+		$path as File\Path;
 
 		$this->writeStatusLine($response);
 		$this->writeHeaders($response);
 
 		$temporary_file_path = $path->toString();
 
-		await $body->closeAsync();
-	  await using ($read_handle = Filesystem\open_read_only($temporary_file_path)) {
+		if ($body is IO\NonDisposableHandle) {
```

Comment:
@fredemmott anything better we can do here?

The `$body` is declared as a generic handle in https://github.com/hhvm/hack-http-request-response-interfaces, should we update that to explicitly use either Disposable or NonDisposable handle?

Reply:
Yeah, that should probably be non-disposable; given we're passing it around within other objects, it can't really be disposable.